### PR TITLE
Ported topic watcher to modular 

### PR DESCRIPTION
**Pull request by riley206-pnnl**
Ported topic watcher to modular

### CODE REVIEW

File: src/topic_watcher/agent.py [hunk 1]
```diff
@@ -23,20 +23,18 @@
...
 from volttron import utils
 from volttron.client.known_identities import PLATFORM_TOPIC_WATCHER
-from volttron.client.messaging.health import Status, STATUS_BAD, STATUS_GOOD
-from volttron.client.vip.agent import Agent, Core, RPC
-from volttron.client.vip.agent import build_agent
+from volttron.client.messaging.health import STATUS_BAD, STATUS_GOOD, Status
+from volttron.client.vip.agent import RPC, Agent, Core, build_agent
 from volttron.utils.scheduling import periodic
 from volttron.utils.time import get_aware_utc_now
+from zmq import ZMQError
```
Comment (craigpnnl) [inline src/topic_watcher/agent.py]: This can't rely on zmq at all

**Review by craigpnnl — CHANGES_REQUESTED** [review]

**Comment by craigpnnl** [comment]
@riley206 did you remove the zmq references...

**Comment by riley206-pnnl** [comment]
> @riley206 did you remove the zmq references...

Sorry, I totally missed that.

**Comment by schandrika** [comment]
@riley206 Please update tests to install agent from local dir not from pip.  Currently I cannot run the tests at all as there is no topic watcher in pypi. But even after we push the agent to pypi, test should install and test local copy.
Instead of pip install volttron-topic-watcher, do pip install <relative-path>/volttron-topic-watcher. Get path using current file location, for example - agent_path = Path(__file__).parents[1]

**Comment by schandrika** [comment]
@riley206  Also please look at pep8 warnings. At least some of the them can be easily fixed - unused import, wrong indent, unreachable code etc

**Comment by schandrika** [comment]
instead of using non existent " utils.is_secure_mode()"  use ClientContext.is_secure_mode() 
this way you can also get rid of the "if False" and hence the unreachable code

**Comment by schandrika** [comment]
missing docs - please add docs

**Comment by riley206-pnnl** [comment]
> @riley206 Also please look at pep8 warnings. At least some of the them can be easily fixed - unused import, wrong indent, unreachable code etc

I ran it through the pep8 linter black.

**Comment by riley206-pnnl** [comment]
> instead of using non existent " utils.is_secure_mode()" use ClientContext.is_secure_mode() this way you can also get rid of the "if False" and hence the unreachable code

do I need to import something for this?

Edit - I found the correct import.

**Review by schandrika — CHANGES_REQUESTED** [review]
1. agent.py hardcodes vip id to platform.topic_watcher. This will make code fail on start if the default vip id is used
2. Missing config.py in docs/source. without this you can't run make html locally. See postgresql historian. you can copy paste code from there
3. Update to newer version of sphinx - again refer to postgresql historian

**Comment by schandrika** [comment]
@riley206 Could you test the scenario where you install the agent without config and add config through config store. The config updates doesn't seem to have any effect even after restart. Agent works only when installed with config. Could you please test this use case

**Pull request merged**